common/js/src/executable-module/emscripten-module.js

```diff
@@ -146,6 +146,52 @@ EmscriptenModule.prototype.load_ = async function() {
   this.messageChannel_.onModuleCreated(this.googleSmartCardModule_);
 };
 
+/**
+ * Constructs the settings object for the Emscripten module instance.
+ * @return {!Object}
+ * @private
+ */
+EmscriptenModule.prototype.getEmscriptenApiModuleSettings_ = function() {
+  return {
+    'onAbort': () => {
+      this.onModuleCrashed_();
+    },
+    'print': (message) => {
+      this.onModuleStdoutReceived_(message);
+    },
+    'printErr': (message) => {
+      this.onModuleStderrReceived_(message);
+    },
+  };
+};
+
+/**
+ * Handles the Emscripten module's crash.
+ * @private
+ */
+EmscriptenModule.prototype.onModuleCrashed_ = function() {
+  this.logger_.severe('The Emscripten module crashed');
+  this.dispose();
+};
+
+/**
+ * Handles an stdout message received from the Emscripten module.
+ * @param {string} message
+ * @private
+ */
+EmscriptenModule.prototype.onModuleStdoutReceived_ = function(message) {
+  this.fromModuleMessagesLogger_.info(message);
+};
+
+/**
+ * Handles an stderr message received from the Emscripten module.
+ * @param {string} message
+ * @private
+ */
+EmscriptenModule.prototype.onModuleStderrReceived_ = function(message) {
+  this.fromModuleMessagesLogger_.info(message);
```

Comment:
nit: Is info appropriate for a message from stderr?

Reply:
Currently, all macros like `GOOGLE_SMART_CARD_LOG_` emit messages to stderr, not stdout: https://github.com/GoogleChromeLabs/chromeos_smart_card_connector/blob/11f3044fc54071654bcf3ea0fce48aa14d6ca2be/common/cpp/src/google_smart_card_common/logging/logging.cc#L144
So with the current approach we do need to log stderr by default.

Having said that, I'm planning to eventually adopt a more fine-grained approach that we had for NaCl in the past, which was to send the log messages to JavaScript as form of tuples `(log_severity, log_message)`. After doing that, we'll be able to reestimate how useful stderr logs are.

Let me add a TODO into the CL to point this out.